server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RemoteDelivery.java

```diff
@@ -125,7 +125,18 @@
  * javax <i>mail.smtp.starttls.enable</i> property. Depending on how strict your security policy is, you might consider
  * <i>mail.smtp.starttls.required</i> as well. Be aware that configuring trust will then be required.<br/>
  * Read <a href="https://javaee.github.io/javamail/docs/api/com/sun/mail/smtp/package-summary.html"><code>com.sun.mail.smtp</code></a>
- * for full information.
+ * for full information.<br/>
+ * <br/>
+ * <b>Mail encoding:</b><br/>
+ * By enabling the javax mail property `mail.smtp.allow8bitmime` to true. James RemoteDelivery tries to convert quoted-printable
+ * or base64 encoding content into 8bit encoding if the remote servers supports 8BITMIME extension.
+ * See {@link org.apache.james.transport.mailets.remote.delivery.RemoteDeliveryConfiguration#createFinalJavaxProperties}
+ * <br/>
+ * This behavior leads to the result of altering mail contents and `Content-Transfer-Encoding` header. In case you are
+ * configuring RemoteDelivery with some other mailets that are sensitive with mail altering,
+ * please override `mail.smtp.allow8bitmime` to false.
+ * <br/>
+ * One example is the RemoteDelivery can break DKIM signatures signed by DKIMSign mailet.
```

Comment:
what about writing an integration test showing that?

Reply:
Oh shit, I thought it was easy but I am wrong now. The mail is copied and altered inside SMTPTransport, no way we can get it through Mailet pipeline. The only possible way is get the mail from remote servers. FakeSMTP doesn't support, MockSMTPServer need to be added one more API to download .eml files

Reply:
added